### PR TITLE
FFWEB-1759: Add generic class for exporting custom columns

### PR DESCRIPTION
**Pull request by a-laurowski**
- Description: 
Introduced way for defining new columns for simple attributes, which doesn't require to utilize any other logic than retrieving from product. Additionally errors thrown there are logged to separate log file.
- Tested with Magento editions/versions: 
2.3.2
- Tested with PHP versions: 
7.2

**Please note that the source and target branch must be `master` ([details](https://github.com/FACT-Finder-Web-Components/magento2-module/blob/HEAD/.github/CONTRIBUTING.md)).**

### CODE REVIEW

File: src/Model/Export/Catalog/ProductField/GenericField.php [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    /**
+     * @param Product $product
+     *
+     * @return string
+     * @throws AttributeExportException
+     */
+    public function getValue(Product $product): string
+    {
+        try {
+            return (string) $product->getAttributeText($this->attributeName);
```
Comment (fmarangi) [inline src/Model/Export/Catalog/ProductField/GenericField.php:45]: This one throws a Fatal Error, i don't think it's catchable. I'd rather check for attribute existence. Also, we have to mimic what we already do for the Attributes field, i wouldnt restrict it to attributes with option.

File: src/Model/Export/Catalog/ProductField/GenericField.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        } catch (\Throwable $e) {
+            if ($this->scopeConfig->isSetFlag('factfinder/general/logging_enabled')) {
```
Comment (fmarangi) [inline src/Model/Export/Catalog/ProductField/GenericField.php:47]: I'd rather just throw the exception and catch it for example in the export script, i dont think logging belongs in this class

Comment (a-laurowski) [reply src/Model/Export/Catalog/ProductField/GenericField.php]: I used the attribute repository. It throws NoSuchEntityException if attribute doesn't exist. With that, i think the Custom exception class i added is redundant.

File: src/Model/Export/Catalog/ProductField/GenericField.php (hunk 1 above)
Comment (a-laurowski) [reply src/Model/Export/Catalog/ProductField/GenericField.php]: i moved whole switch instruction from `Attributes` to separate class and use it also in `GenericField`. Now it should be capable of exporting different types of attributes.

**Pull request merged**